src/foam/dao/Relationship.js

```diff
@@ -254,18 +257,21 @@ foam.CLASS({
         foam.RELATIONSHIP({
           sourceModel: this.targetModel,
           targetModel: id,
+          targetProperty: this.targetProperty,
           forwardName: this.inverseName,
           inverseName: 'targetId',
           sourceDAOKey: this.targetDAOKey,
           targetDAOKey: this.junctionDAOKey,
-          relationshipDAOFactory: function(source) {
+          relationshipDAOFactory: function(s) {
             return foam.dao.ManyToManyRelationshipDAO.create({
-              obj: source,
+              dest: jModel,
+              obj: s,
+              of: source.id,
```

Comment:
You renamed "source" to "s" but didn't change it on this line.  Why not just leave it as "source" so the diff is smaller.